matcher/core.py

```diff
@@ -36,6 +36,7 @@ def __init__(
                 maximums=[],
                 demands=[],
                 num_alternates=0,
+                allow_only_pos_affinity=False,
```

Comment:
I would like the default behavior to be "not allow the 0 score assignments". This is going to be the most used option and there will be exceptionally cases when we want to allow assignments with 0 scores. Can we rename it to
`allow_zero_score_assignments=False`?

Reply:
@ameyagodbole do you want to take this up?